alpha/apps/kaltura/lib/kPermissionManager.php

```diff
@@ -702,6 +706,96 @@ private static function initPermissionsMap()
 				// merge current role map to the global map
 				self::$map = array_merge_recursive(self::$map, $roleMap);
 			}
+
+			self::removeLimitedPermissions();
+		}
+	}
+
+	private static function removeLimitedPermissions()
+	{
+		$ks = ks::fromSecureString(kCurrentContext::$ks);
+		if (!$ks)
+		{
+			return;
+		}
+		$limitRole = $ks->getRole(kSessionBase::PRIVILEGE_LIMIT_ROLE);
+		if (!$limitRole)
+		{
+			return;
+		}
+
+		KalturaLog::debug("Removing permissions according to limit role [$limitRole]");
+		$roleMap = self::getPermissions($limitRole, true);
+
+		self::removeApiActionPermission($roleMap[self::API_ACTIONS_ARRAY_NAME]);
+		self::removeApiParametersPermissions($roleMap[self::API_PARAMETERS_ARRAY_NAME]);
+		self::removePermissionNames($roleMap[self::PERMISSION_NAMES_ARRAY]);
+	}
+
+	private static function removeApiActionPermission($apiActions)
+	{
+		foreach ($apiActions as $service => $actionsArray)
+		{
+			if (!isset(self::$map[self::API_ACTIONS_ARRAY_NAME][$service]))
+			{
+				continue;
+			}
+			foreach ($actionsArray as $actionName => $actionEmptyArray)
+			{
+				KalturaLog::debug("Removing permission for service [$service] action [$actionName]");
```

Comment:
makes more sense to have it after the if, as only on this case you remove it
same for the other 2 functions as well